File: option.cpp
#include <iostream> // cout, endl
#include <stdlib.h> // rand
#include <time.h> // time
using namespace std;

// If there is a 'random error' returns NULL
const int* const may_return_null () {
  srand(time(NULL));
  return rand() % 2 > 0 ? new int(666) : NULL;
}

int main () {
  // if else
  const int* const x = may_return_null();
  if (x) {
    // switch
    switch (*x) {
      case 777: cout << "Lucky Sevens"        << endl; break;
      case 666: cout << "Number of the Beast" << endl; break;
      case 42: cout  << "Meaning of Life"     << endl; break;
      default: cout  << "Nothing special"     << endl; break;
    }
  } else {
    cout << "No value" << endl;
  }
  
  // single if
  if (*x == 666) {
    cout << "Did I mention that Iron Maiden is my favorite band?" << endl;
  }
}

File: option.rs
use core::rand;

fn may_return_none () -> Option<@int> {
  if rand::Rng().next() % 2 == 1 { Some(@666) } else { None }
}

fn main () {
  let x: Option<@int> = may_return_none();
  
  io::println(match x {
    Some(y) if *y == 777 => "Lucky Sevens",
    Some(y) if *y == 666 => "Number of the Beast",
    Some(y) if *y == 42 => "Meaning of Life",
    Some(_) => "Nothing special",
    None => "No value"
  });
  
  match x {
    Some(y) if *y == 666 => {
      io::println("Did I mention that Iron Maiden is my favorite Band?")
    },
    _ => {}
  };
}
